applications/nrf5340_audio/src/bluetooth/bt_stream/unicast/unicast_client.c

```diff
@@ -1597,25 +1668,46 @@ int unicast_client_stop(void)
 	return 0;
 }
 
-int unicast_client_send(struct le_audio_encoded_audio enc_audio)
+int unicast_client_send(uint8_t cig_index, struct le_audio_encoded_audio enc_audio)
 {
 #if (CONFIG_BT_AUDIO_TX)
 	int ret;
-	struct bt_bap_stream *bap_tx_streams[CONFIG_BT_BAP_UNICAST_CLIENT_ASE_SNK_COUNT];
-	uint8_t audio_mapping_mask[CONFIG_BT_BAP_UNICAST_CLIENT_ASE_SNK_COUNT] = {UINT8_MAX};
-
-	for (int i = 0; i < CONFIG_BT_BAP_UNICAST_CLIENT_ASE_SNK_COUNT; i++) {
-		bap_tx_streams[i] = &unicast_servers[i].cap_sink_stream.bap_stream;
-		if (unicast_servers[i].location == BT_AUDIO_LOCATION_FRONT_RIGHT) {
-			audio_mapping_mask[i] = AUDIO_CH_R;
-		} else {
-			/* Both mono and left unicast_servers will receive left channel */
-			audio_mapping_mask[i] = AUDIO_CH_L;
+	uint8_t num_active_streams = 0;
+	struct le_audio_tx_info tx[ARRAY_SIZE(unicast_servers[cig_index][0])];
+
+	for (int i = 0; i < ARRAY_SIZE(unicast_servers[cig_index][0]); i++) {
+		/* Skip unicast_servers not in a streaming state */
+		if (!le_audio_ep_state_check(
+			    unicast_servers[cig_index][0][i].cap_sink_stream.bap_stream.ep,
+			    BT_BAP_EP_STATE_STREAMING)) {
+			continue;
 		}
+
+		/* Set cap stream pointer */
+		tx[num_active_streams].cap_stream =
+			&unicast_servers[cig_index][0][i].cap_sink_stream;
+
+		/* Set index */
+		tx[num_active_streams].idx.lvl1_idx = cig_index;
+		tx[num_active_streams].idx.lvl2_idx = 0;
+		tx[num_active_streams].idx.lvl3_idx = i;
+
+		/* Set channel location */
+		/* Both mono and left unicast_servers will receive left channel */
+		tx[num_active_streams].audio_channel =
+			(unicast_servers[cig_index][0][i].location == BT_AUDIO_LOCATION_FRONT_RIGHT)
```

Comment:
Could this be rewritten to check for FRONT_LEFT? It seems more intuitive. Personally, I find the ternary operator a bit more challenging to read.

Reply:
No, since we need to set all devices that are NOT FRONT_RIGHT to AUDIO_CH_L we need to test for BT_AUDIO_LOCATION_FRONT_RIGHT.
Yes, it might be a bit harder to read but boy does it make the code neater. If we used it more it would be easier to read when they do pop up ;)